lib/ebay_api/middlewares/save_to_file_response.rb

```diff
@@ -0,0 +1,29 @@
+require "securerandom"
+
+# Parses gzipped response
+class EbayAPI
+  class SaveToFileResponse
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      status, headers, body = @app.call(env)
+
+      [status, headers, body.map { |b| save(b) }]
+    end
+
+    private
+
+    def save(body)
+      return unless body
+      return if body.empty?
+
+      file = File.new("/tmp/ebay-api-response-#{SecureRandom.hex}", "w+b")
```

Comment:
Maybe use [Tempfile](https://ruby-doc.org/stdlib-2.5.3/libdoc/tempfile/rdoc/Tempfile.html) here?

```suggestion
      file = Tempfile.new("ebay-api-response", binmode: true)
```

Reply:
Well, it was my first decision to use it but `Tempfile` is [automatically removed](https://ruby-doc.org/stdlib-3.0.3/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close) when the object is cleaned by GC. So, although it looks buggy, I decided to use plain `File` instead, and I ensure removal of it in the caller.